src/app.tsx

```diff
@@ -108,6 +110,12 @@ export const Application = () => {
 
     useEffect(
         () => {
+            // Don't render when there is no user, this avoids showing `/`
+            // briefly and then loading the users home direcotry.
+            if (user === null) {
+                return;
+            }
```

Comment:
I find this really weird -- the code here in this effect doesn't actually need `user` -- or at least it didn't directly refer to it. Is that used indirectly? If the effect does use it, ok -- but if not, let's just fix this in the render method (i.e. render nothing until user is initialized). TBH I would even pull this out completely, and only instantiate `<Application>` after cockpit.user() finishes. There is very little sense to treat this as state.